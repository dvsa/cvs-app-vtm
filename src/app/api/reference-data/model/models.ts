export * from './reference-data.model';
