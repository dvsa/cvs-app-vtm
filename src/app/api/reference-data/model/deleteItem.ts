/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}