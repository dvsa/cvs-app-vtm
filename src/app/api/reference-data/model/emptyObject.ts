/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
