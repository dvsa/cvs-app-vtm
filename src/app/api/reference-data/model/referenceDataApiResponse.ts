/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ReferenceDataApiResponseWithPagination } from './referenceDataApiResponseWithPagination';
import { ReferenceDataApiResponseWithoutPagination } from './referenceDataApiResponseWithoutPagination';

export type ReferenceDataApiResponse = ReferenceDataApiResponseWithoutPagination | ReferenceDataApiResponseWithPagination;