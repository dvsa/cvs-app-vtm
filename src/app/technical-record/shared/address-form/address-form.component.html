<section [formGroup]="addressFg" class="govuk-!-margin-top-6 govuk-!-margin-bottom-6">
  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-address1">
      Building and street
    </label>
    <input
      id="test-edit-address1"
      type="text"
      formControlName="address1"
      class="govuk-input govuk-input--width-20"
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-address2"> </label>
    <input
      id="test-edit-address2"
      type="text"
      formControlName="address2"
      class="govuk-input govuk-input--width-20"
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-postTown">
      Town or city
    </label>
    <input
      id="test-edit-postTown"
      type="text"
      formControlName="postTown"
      class="govuk-input govuk-input--width-20"
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-address3">
      County (optional)
    </label>
    <input
      id="test-edit-address3"
      type="text"
      formControlName="address3"
      class="govuk-input govuk-input--width-20"
      vtmNullValue
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-psotCode">
      Postcode (optional)
    </label>
    <input
      id="test-edit-postCode"
      type="text"
      formControlName="postCode"
      class="govuk-input govuk-input--width-20"
      vtmNullValue
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-telephoneNumber">
      Telephone number (optional)
    </label>
    <input
      id="test-edit-telephoneNumber"
      type="text"
      formControlName="telephoneNumber"
      class="govuk-input govuk-input--width-20"
      vtmNullValue
    />
  </div>

  <div class="govuk-form-group">
    <label class="govuk-label govuk-!-font-weight-bold" for="test-edit-emailAddress">
      Email address (optional)
    </label>
    <input
      id="test-edit-emailAddress"
      type="text"
      formControlName="emailAddress"
      class="govuk-input govuk-input--width-20"
      vtmNullValue
    />
  </div>
</section>
