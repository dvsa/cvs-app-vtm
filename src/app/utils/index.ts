export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


