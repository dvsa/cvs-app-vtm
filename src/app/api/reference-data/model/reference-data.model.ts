import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
