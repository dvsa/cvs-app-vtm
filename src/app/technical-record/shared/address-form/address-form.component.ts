import { Component } from '@angular/core';
import { FormGroup, FormBuilder } from '@angular/forms';
import { AddressInformation } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-address-form',
  templateUrl: './address-form.component.html'
})
export class AddressFormComponent {
  addressFg: FormGroup;

  constructor(private fb: FormBuilder) {}

  createControls(entity: AddressInformation) {
    this.addressFg = this.fb.group({
      address1: this.fb.control(entity.address1),
      address2: this.fb.control(entity.address2),
      postTown: this.fb.control(entity.postTown),
      address3: this.fb.control(entity.address3),
      postCode: this.fb.control(entity.postCode),
      telephoneNumber: this.fb.control(entity.telephoneNumber),
      emailAddress: this.fb.control(entity.emailAddress)
    });

    return this.addressFg.controls;
  }
}
