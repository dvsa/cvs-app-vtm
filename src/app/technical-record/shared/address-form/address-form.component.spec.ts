import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AddressFormComponent } from './address-form.component';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { AddressInformation } from '@app/models/tech-record.model';

describe('AddressFormComponent: ', () => {
  let component: AddressFormComponent;
  let fixture: ComponentFixture<AddressFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule],
      declarations: [AddressFormComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AddressFormComponent);
    component = fixture.componentInstance;
  });

  it('should create controls with default entity', () => {
    component.createControls({} as AddressInformation);
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  it('should create controls with populated entity', () => {
    const mockAddressInfo = TESTING_UTILS.mockAddressInformation();

    component.createControls(mockAddressInfo);
    expect(component.addressFg.get('address1').value).toEqual(mockAddressInfo.address1);
    expect(component.addressFg.get('address2').value).toEqual(mockAddressInfo.address2);
    expect(component.addressFg.get('address3').value).toEqual(mockAddressInfo.address3);
    expect(component.addressFg.get('postTown').value).toEqual(mockAddressInfo.postTown);
    expect(component.addressFg.get('postCode').value).toEqual(mockAddressInfo.postCode);
    expect(component.addressFg.get('emailAddress').value).toEqual(mockAddressInfo.emailAddress);
    expect(component.addressFg.get('telephoneNumber').value).toEqual(
      mockAddressInfo.telephoneNumber
    );
  });
});
