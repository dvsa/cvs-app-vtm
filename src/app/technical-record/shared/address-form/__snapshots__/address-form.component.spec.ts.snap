// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressFormComponent:  should create controls with default entity 1`] = `
<vtm-address-form
  addressFg={[Function FormGroup]}
  fb={[Function FormBuilder]}
>
  <section
    class="govuk-!-margin-top-6 govuk-!-margin-bottom-6 ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
  >
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-address1"
      >
         Building and street 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="address1"
        id="test-edit-address1"
        ng-reflect-name="address1"
        type="text"
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-address2"
      />
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="address2"
        id="test-edit-address2"
        ng-reflect-name="address2"
        type="text"
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-postTown"
      >
         Town or city 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="postTown"
        id="test-edit-postTown"
        ng-reflect-name="postTown"
        type="text"
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-address3"
      >
         County (optional) 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="address3"
        id="test-edit-address3"
        ng-reflect-name="address3"
        type="text"
        vtmnullvalue=""
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-psotCode"
      >
         Postcode (optional) 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="postCode"
        id="test-edit-postCode"
        ng-reflect-name="postCode"
        type="text"
        vtmnullvalue=""
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-telephoneNumber"
      >
         Telephone number (optional) 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="telephoneNumber"
        id="test-edit-telephoneNumber"
        ng-reflect-name="telephoneNumber"
        type="text"
        vtmnullvalue=""
      />
    </div>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-!-font-weight-bold"
        for="test-edit-emailAddress"
      >
         Email address (optional) 
      </label>
      <input
        class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
        formcontrolname="emailAddress"
        id="test-edit-emailAddress"
        ng-reflect-name="emailAddress"
        type="text"
        vtmnullvalue=""
      />
    </div>
  </section>
</vtm-address-form>
`;
